src/main/java/seedu/duke/manager/Manager.java

```diff
@@ -0,0 +1,96 @@
+package seedu.duke.manager;
+
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Manager class provides the base functionality that all Managers in the application must implement.
+ * Managers function as the 'Controller' in the application.
+ *
+ * <p>Provides base implementation and helper methods to load data from database files.
+ */
+public abstract class Manager {
+    /**
+     * Database subdirectory name.
+     */
+    private static final String DATABASE_DIRECTORY = "data";
+    /**
+     * Data storage location.
+     */
+    private final String file;
+
+    /**
+     * Creates a Manager.
+     *
+     * @param file Name of the database file to store your required data (i.e. "employee.dat").
+     */
+    protected Manager(String file) {
+        this.file = Path.of(DATABASE_DIRECTORY, file).toString();
+    }
+
+    /**
+     * Ensure that the folder to store the database files are generated.
+     *
+     * @throws IOException Error when creating folder.
+     */
+    private void ensureDatabaseDirectoryExists() throws IOException {
+        Files.createDirectories(Path.of(DATABASE_DIRECTORY));
+    }
+
+    /**
+     * Wrapper to help with loading of database.
+     *
+     * @return Object.
+     * @throws Exception For any errors while reading.
+     */
+    protected Object load() throws Exception {
+        this.ensureDatabaseDirectoryExists();
+
+        FileInputStream fis = new FileInputStream(this.file);
```

Comment:
Here should be the same stream for different calls?

Reply:
u shud code it such that it only loads once when you open the app, and only saves once when u exit the app nicely

Reply:
ah i see, thought i needed to read one by one